### DISCUSSION TITLE
Use extension in dev environment with extension improvement branch

### DISCUSSION THREAD

**Discussion by VendoAU** [Minestom/Minestom#790] [Help]
Hi, I am trying to use an extension without building a jar each time I want to update it, previously I did it like [this](https://wiki.minestom.net/expansion/extensions#testing-in-a-dev-environment) but that doesn't seem to work in the extension improvement branch.
The error that I am getting is this:
`Unable to find or load main class 'com.vendoau.ctf.CaptureTheFlag' in extension 'CaptureTheFlag'.: java.lang.ClassNotFoundException: com.vendoau.ctf.CaptureTheFlag`

This is my extension.json:
```json
{
  "entrypoint": "com.vendoau.ctf.CaptureTheFlag",
  "name": "CaptureTheFlag",
  "version": "1.0.0"
}
```

**Comment by mworzala** [comment]
I can reproduce. I am not immediately sure the cause of the issue. I'll update here when it is fixed. A hacky workaround is to give the path to the extension jar to `-Dminestom.extension.indevfolder.classes`, then build the jar instead of just the classes. 

I would suggest using the "autoload" mechanism added in this branch instead however. If you put the extension on the classpath of the server, the extension will be loaded on its own and can be accessed by the debugger (eg hot reloaded, breakpoints, etc).
